src/builder/Actions.sol

```diff
@@ -541,6 +683,138 @@ library Actions {
         return (quarkOperation, action);
     }
 
+    function unwrapAsset(UnwrapAsset memory unwrap, PaymentInfo.Payment memory payment, bool useQuotecall)
+        internal
+        pure
+        returns (IQuarkWallet.QuarkOperation memory, Action memory)
+    {
+        bytes[] memory scriptSources = new bytes[](1);
+        scriptSources[0] = type(WrapperActions).creationCode;
+
+        Accounts.ChainAccounts memory accounts = Accounts.findChainAccounts(unwrap.chainId, unwrap.chainAccountsList);
+
+        Accounts.AssetPositions memory assetPositions =
+            Accounts.findAssetPositions(unwrap.assetSymbol, accounts.assetPositionsList);
+
+        Accounts.QuarkState memory accountState = Accounts.findQuarkState(unwrap.sender, accounts.quarkStates);
+
+        // Construct QuarkOperation
+        IQuarkWallet.QuarkOperation memory quarkOperation = IQuarkWallet.QuarkOperation({
+            nonce: accountState.quarkNextNonce,
+            scriptAddress: CodeJarHelper.getCodeAddress(type(WrapperActions).creationCode),
+            scriptCalldata: TokenWrapper.encodeUnwrapToken(unwrap.chainId, unwrap.assetSymbol, unwrap.amount),
+            scriptSources: scriptSources,
+            expiry: unwrap.blockTimestamp + STANDARD_EXPIRY_BUFFER
+        });
+
+        if (payment.isToken) {
+            // Wrap operation with paycall
+            quarkOperation = useQuotecall
+                ? QuotecallWrapper.wrap(
+                    quarkOperation, unwrap.chainId, payment.currency, PaymentInfo.findMaxCost(payment, unwrap.chainId)
+                )
+                : PaycallWrapper.wrap(
+                    quarkOperation, unwrap.chainId, payment.currency, PaymentInfo.findMaxCost(payment, unwrap.chainId)
+                );
+        }
+
+        // Construct Action
+        WrappingActionContext memory wrapActionContext = WrappingActionContext({
```

Comment:
Was trying to find a mutual usable name for the wrapping action context, cuz the context itself we can just use one as from and to token types. For wrap and unwrap instead of having two highly similar struct in the code.

Reply:
Renamed this to `WrapOrUnwrapActionContext`